cli/azd/pkg/project/framework_service_dotnet.go

```diff
@@ -42,16 +42,27 @@ func (dp *dotnetProject) Package(ctx context.Context, progress chan<- string) (s
 }
 
 func (dp *dotnetProject) InstallDependencies(ctx context.Context) error {
-	dotnetCli := tools.NewDotNetCli()
-	if err := dotnetCli.Restore(ctx, dp.config.Path()); err != nil {
+	if err := dp.dotnetCli.Restore(ctx, dp.config.Path()); err != nil {
+		return err
+	}
+	return nil
+}
+
+func (dp *dotnetProject) Initialize(ctx context.Context) error {
+	if err := tools.EnsureInstalled(ctx, dp.dotnetCli); err != nil {
```

Comment:
Let's remove this check from here - the other implementations don't do this and we'll enforce this elsewhere (via the `RequiredExternalTools` method)